nltk/tag/stanford.py

```diff
@@ -204,6 +205,75 @@ def parse_output(self, text, sentences):
 
         raise NotImplementedError
 
+class CoreNLPTagger(CoreNLPParser, TaggerI):
+    def __init__(self, tagtype, url='http://localhost:9000', encoding='utf8'):
+        """
+        An abstract interface to POS/NER taggers of CoreNLP that returns the
+        POS/NER tags from the Stanford CoreNLP API at nltk.parse.corenlp.
+        """
+        self.tagtype = tagtype
+        super(CoreNLPTagger, self).__init__(url, encoding)
+
+    def tag_sents(self, sentences):
+        sentences = (' '.join(words) for words in sentences)
+        return list(self.raw_tag_sents(sentences))
+
+
+    def tag(self, sentence):
+        return self.tag_sents([sentence])[0]
+
+    def raw_tag_sents(self, sentences):
+        """
+        This function will interface the `GenericCoreNLPParser.api_call` to
+        retreive the JSON output and return the annotations required.
+        """
+        default_properties = {'ssplit.isOneSentence': 'true',
+                              'annotators': 'tokenize,ssplit,' }
+        # Supports only 'pos' or 'ner' tags.
+        assert self.tagtype in ['pos', 'ner']
+        default_properties['annotators'] += self.tagtype
+        for sentence in sentences:
+            tagged_data = self.api_call(sentence, properties=default_properties)
+            assert len(tagged_data['sentences']) == 1
+            # Taggers only need to return 1-best sentence.
+            yield [(token['word'], token[self.tagtype]) for token in tagged_data['sentences'][0]['tokens']]
+
+
+class CoreNLPPOSTagger(CoreNLPTagger):
+    """
+    This is a subclass of the CoreNLPTagger that wraps around the
+    nltk.parse.CoreNLPParser for Part-of-Sppech tagging.
+
+        >>> from nltk.tag.stanford import CoreNLPPOSTagger
+        >>> tagged = CoreNLPPOSTagger().tag('What is the airspeed of an unladen swallow ?'.split()) # doctest: +SKIP
+        >>> expected = [('What', 'WP'), ('is', 'VBZ'), ('the', 'DT'),
+        ... ('airspeed', 'NN'), ('of', 'IN'), ('an', 'DT'), ('unladen', 'JJ'),
+        ... ('swallow', 'VB'), ('?', '.')]
+        >>> expected == tagged # doctest: +SKIP
+        True
+    """
+    def __init__(self, url='http://localhost:9000', encoding='utf8'):
```

Comment:
It's redefining the `__init__` here because `CoreNLPTagger` the `self.tagtype` needs to be set by the subclass of `CoreNLPTagger`. The functions are handled by the superclass. Optionally, we can drop the `__init__` in `CoreNLPPOSTagger` and `CoreNLPNERTagger` but we have to override the `raw_tag_sents` to handle the `self.tagtype` differences. 
 
Another way is to use `*args` and `**kwargs` but that'll give too much freedom to the parameters and I think explicit parameters is easy to read. 

Maybe there's another way to access self.tagtype from the subclass in the superclass' `raw_tag_sents`. Any ideas?